src/main/java/com/kryptnostic/kindling/search/ConductorElasticsearchImpl.java

```diff
@@ -0,0 +1,324 @@
+				.should( QueryBuilders.matchQuery( ENTITY_SET + "." + DESCRIPTION, searchTerm ).fuzziness( Fuzziness.AUTO ) )
+				.minimumNumberShouldMatch( 1 );
+		}
+		
+		if ( optionalEntityType.isPresent() ) {
+			UUID eid = optionalEntityType.get();
+			query.must( QueryBuilders.matchQuery( ENTITY_SET + "." + ENTITY_TYPE_ID, eid.toString() ) );
+		} else if ( optionalPropertyTypes.isPresent() ) {
+			Set<UUID> propertyTypes = optionalPropertyTypes.get();
+			for ( UUID pid: propertyTypes ) {
+				query.must( QueryBuilders.nestedQuery( PROPERTY_TYPES, QueryBuilders.matchQuery( PROPERTY_TYPES + "." + ID, pid.toString() ), ScoreMode.Avg ) );
+			}
+		}
+		SearchResponse response = client.prepareSearch( ENTITY_SET_DATA_MODEL )
+				.setTypes( ENTITY_SET_TYPE )
+				.setQuery( query )
+				.setFetchSource( new String[]{ ENTITY_SET, PROPERTY_TYPES }, null )
+				.setFrom( 0 ).setSize( 50 ).setExplain( true )
+				.get();
+		
+		List<Map<String, Object>> hits = Lists.newArrayList();
+		for ( SearchHit hit: response.getHits() ) {
+			Map<String, Object> match = hit.getSource();
+			Set<String> permissions = Sets.newHashSet();
+			for( SearchHits innerHits: hit.getInnerHits().values() ) {
+				for (SearchHit innerHit: innerHits.getHits() ) {
+					permissions.addAll( (List<String>) innerHit.getSource().get( ACLS ) );
+				}
+			}
+			match.put( ACLS, permissions );
+			hits.add( match );
+		}
+		return hits;
+	}
+	
+	@Override
+	public Boolean updateEntitySetPermissions( UUID entitySetId, Principal principal, Set<Permission> permissions ) {
+		try {
+			if ( !verifyElasticsearchConnection() ) return false;
+		} catch (UnknownHostException e) {
+			e.printStackTrace();
+		}
+		Map<String, Object> acl = Maps.newHashMap();
+        acl.put( ACLS, permissions );
+        acl.put( TYPE, principal.getType().toString() );
+        acl.put( NAME, principal.getId() );
+        acl.put( ENTITY_SET_ID, entitySetId.toString() );
+        
+		ObjectMapper mapper = new ObjectMapper();
+		mapper.registerModule( new GuavaModule() );
+		mapper.registerModule( new JodaModule() );
+		try {
+			String s = mapper.writeValueAsString( acl );
+			String id = entitySetId.toString() + "_" + principal.getType().toString() + "_" + principal.getId();
+			client.prepareIndex( ENTITY_SET_DATA_MODEL, ACLS, id ).setParent( entitySetId.toString() ).setSource( s ).execute().actionGet();
+			return true;
+		} catch (JsonProcessingException e) {
+			e.printStackTrace();
+		}
+		return false;
+	}
+	
+	@Override
+	public Boolean updatePropertyTypesInEntitySet( UUID entitySetId, Set<PropertyType> newPropertyTypes ) {
+		try {
+			if ( !verifyElasticsearchConnection() ) return false;
+		} catch (UnknownHostException e) {
+			e.printStackTrace();
+		}
+		
+		Map<String, Object> propertyTypes = Maps.newHashMap();
+		propertyTypes.put( PROPERTY_TYPES, newPropertyTypes);
+		ObjectMapper mapper = new ObjectMapper();
```

Comment:
Use ObjectMappers if you can to get your object mapper. They are expensive to setup and are designed to be used by multiple threads safely.